x-pack/plugins/observability_solution/observability_onboarding/public/plugin.ts

```diff
@@ -76,6 +76,7 @@ export class ObservabilityOnboardingPlugin
 
     const pluginSetupDeps = plugins;
 
+    const isServerless = this.ctx.env.packageInfo.buildFlavor === 'serverless';
```

Comment:
It seems like there are two ways how to check for serverless - via the env like you are doing it here and via `isServerlessEnabled` on the cloud plugin:

https://github.com/elastic/kibana/blob/aaf3ad76acc7ad51d0756a605384f8d200c9d9a5/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/setup_technology.ts#L20

I don't have a strong opinion, but the contract one seems to be better suited as it's explicitly meant for this case while the build flavor might be unreliable.

Reply:
I will ping the serverless platform channel to see if they have guidance on the build flavor flag. This is being used extensively in observability solutions so if we can't rely on it for this purpose we should let people know so they can refactor.